.github/workflows/cherry-pick.yml

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# This action cherry-picks CVE fix commits from fast-track branches to corresponding
+# development branches for each version of Mariner.
+# To use this action, create a comment in a merged PR to any 'fasttrack/*' branch:
+#   /cherry-pick <target-branch-1>, <target-branch-2>, etc
+# The author of the comment must be a member or owner of the repository for the action
+# to work.
+# After cherry-pick succeeds for a branch, a new PR will be created to merge the commit
+# to that target branch, and a label will be added to the original PR to indicate that
+# the commit has been cherry-picked to that branch.
+name: Cherry pick commits from PR
+on:
+  issue_comment:
+    types:
+      - created
+jobs:
+  # Basic information extraction and validation
+  pr_info:
+    if: |
+      github.event.issue.pull_request &&
+        startsWith(github.event.comment.body, '/cherry-pick ') &&
+        (github.event.comment.author_association == 'MEMBER' ||
+         github.event.comment.author_association == 'OWNER')
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    outputs:
+      status: ${{ steps.pr_info.outputs.status }}
+      base_ref: ${{ steps.pr_info.outputs.base_ref }}
+      merge_commit: ${{ steps.pr_info.outputs.merge_commit }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Extract PR information
+        id: pr_info
+        run: |
+          echo "status=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json state --jq '.state')" >> $GITHUB_OUTPUT
+          echo "base_ref=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json baseRefName --jq '.baseRefName')" >> $GITHUB_OUTPUT
+          echo "merge_commit=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json mergeCommit --jq '.mergeCommit.oid')" >> $GITHUB_OUTPUT
+
+      - name: Check that PR satisfies requirements for cherry-pick action and notify if not
+        run: |
+          if [[ "${{ steps.pr_info.outputs.status }}" != "MERGED" ]] || [[ "${{ steps.pr_info.outputs.base_ref }}" != fasttrack/* ]]; then
+            gh pr comment ${{ github.event.issue.number }} \
+              --repo ${{ github.repository }} \
+              --body 'Cherry pick action is only enabled for merged pull requests to `fasttrack/*` branches'
```

Comment:
Can we make it so, that we can add the `/cherry-pick X` comment whenever we want (ever before the PR is merged) but the bot will only execute the cherry-pick after the change gets merged? This should make it harder to forget about the comment.

And ideally: could we cherry-pick any PR merged into the `fast-track/*` branch to its corresponding dev branch? That is the default we're interested in. I wouldn't mind adding an optional comment to skip that cherry-picking for a single PR, though, but that's a P2 at the moment.

Reply:
I updated the workflow. Now it's triggered whenever a PR to fast-track branch is merged. The workflow will cherry-pick to the corresponding dev branch of the fast-track branch by default (specified by the environment variable `CHERRY_PICK_BRANCH_MAPPING`). And we also have an option to specify additional target branches via comments

Reply:
one thing though, is that this action is now triggered on PR merge only, so commenting after merging won't work anymore